src/TrustedCodehashAccess.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { ITrustedCodehashAccess } from "./interfaces/ITrustedCodehashAccess.sol";
+/**
+ * @title TrustedCodehashAccess
+ * @author Ricardo Guilherme Schmidt <ricardo3@status.im>
+ * @notice Ensures that only specific contract bytecode hashes are trusted to
+ *         interact with the functions using the `onlyTrustedCodehash` modifier.
+ */
+
+contract TrustedCodehashAccess is ITrustedCodehashAccess, Ownable {
+    mapping(bytes32 codehash => bool permission) private trustedCodehashes;
+
+    /**
+     * @notice Restricts access based on the codehash of the caller.
+     *         Only contracts with trusted codehashes can execute functions using this modifier.
+     */
+    modifier onlyTrustedCodehash() {
+        bytes32 codehash = msg.sender.codehash;
+        if (!trustedCodehashes[codehash]) {
+            revert TrustedCodehashAccess__UnauthorizedCodehash();
+        }
+        _;
+    }
+
+    constructor(address _owner) Ownable(_owner) { }
```

Comment:
@3esmit FYI: `Ownable` requires its constructor to be called so seems we have to introduce one here